### PR TITLE
Report E704 for async def as well

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

File: testsuite/E70.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 del a[:]; a.append(42);
 #: E704:1:1
 def f(x): return 2
+#: E704:1:1
+async def f(x): return 2
```
Comment (spaceone) [inline testsuite/E70.py]: Does it detect two spaces?
```
#: E704:1:1
+async  def f(x): return 2
```

Comment (asottile) [inline testsuite/E70.py]: indeed it does not, I can extend this further

Comment (asottile) [inline testsuite/E70.py]: I've fixed this case as well as two others where 'async  def' was not being detected

Comment (sigmavirus24) [inline testsuite/E70.py]: In my opinion 2 spaces is not something that should be caught. That should trigger a different warning, probably along the lines of the warnings that are triggered when there are too many spaces around a `=`.

Comment (asottile) [inline testsuite/E70.py]: It also triggers that, but now triggers both

Comment (asottile) [inline testsuite/E70.py]: I think it's beneficial to trigger both warnings -- it reduces the number of steps to finish refactoring some bad code.

With this patch it immediately shows both warnings, but without this patch you would first see E272, fix it, and then see E704.

Comment (asottile) [inline testsuite/E70.py]: Or if someone (for some reason) disabled E272 but not E704

Comment (sigmavirus24) [inline testsuite/E70.py]: Good point. That works for me. :)

**Pull request merged**